### ISSUE TITLE
Is this package in an usable state?

### ISSUE THREAD

**Issue by onekk** [Aerijo/ide-latex#1]
Hi, is this package in an usable state, I'm using atom-latex, but I'm not very satisfied on spellchecking and error, so I've resolved to install linter-spell-latex and ssome other packages.

If not could you gave some hints on achieving two of my most annoying problems?

- exclude commented code (preceded with %) to be spellchecked by linter-spell-latex
- add latex commands to syntax and exclede them to spellcheck, now many costruct like:

`\begin{whatever is not seen as error}[this part is marked as error by the spellchecker]`

Many thankk for your work, TIA and Regards

Carlo D.

**Comment by Aerijo** [comment]
No, turned out a language server, [texlab](https://texlab.netlify.app/), already existed and was pretty much feature complete.

You might want to try it instead. I've got [a package](https://atom.io/packages/ide-latex-texlab) that will run it for you if you point it to a texlab executable. Just download the [release](https://github.com/latex-lsp/texlab/releases) for your platform, unzip it, and add the server executable path to the package config entry.

**Comment by onekk** [comment]
Thanks, I've installed texlab on my arch linux distribution.

and you package ide-latex-texlab, but it seem not to work, any hint?
 
TIA

Carlo D.

**Comment by Aerijo** [comment]
Have you configured the package to find it? It's been a while, so I don't know exactly how it will search for the server. That's why I recommend to add the absolute path to the server directly (not just it's folder). E.g.,
![image](https://user-images.githubusercontent.com/22167388/102015362-f1787180-3da6-11eb-85f3-abd648a4614c.png)

Should work fine after that, just tried it myself. Check by typing a `\` in a .tex file; the autocomplete should pop up.

**Comment by onekk** [comment]
Yes my executable is in \usr\bin\texlab

so I've added it to the server field

but it don't seem to work.

i.e the file is not recognized as Latex in atom the small status line at the bottom say plain text.

I have uninstalled all the latex related packages, maybe I'm doing something wrong?

Thanks and Regards

Carlo D.

**Comment by Aerijo** [comment]
Do you have a LaTeX syntax highlighting package? Atom recognising a file as LaTeX is required for it to start the language server.

**Comment by onekk** [comment]
OK I've installed your language-latex2e and now it is working.

Is spellchecking managed by TexLab and how to buidl the latex file, istructions are about VCStudio.

I want to use arara 

Many thanks 

Carlo D.

**Comment by Aerijo** [comment]
Sorry, I don't know anything about texlab internals. The language server protocol is (supposed to be) a generic way for language analysis servers and clients like text editors to communicate. Something specific like building a tex file would be a custom behaviour.

I recommend giving VS Code a go, as it's LaTeX support will work much better out of the box (well, with a couple of packages). The texlab maintainers are the ones that maintain the vscode texlab package, so it will be capable of anything texlab is.

Otherwise sticking to Atom will require looking through packages and possibly implementing what you want yourself (or at least 'gluing' the CLI tools to Atom commands). I personally use the [`latex`](https://atom.io/packages/latex) package to compile (with latexmk) and don't use a spell checker, so I can help any further with that. I don't believe the `latex` package has arara support though, and it is not being actively developed so it's unlikely to ever get it.

**Comment by onekk** [comment]
Many thanks.

OK maybe I will give vscode a try.
